lib/email.tsx

```diff
@@ -0,0 +1,67 @@
+import SendGrid from "@sendgrid/mail";
+import WelcomeEmail from "@/emails/welcome-email";
+import { render } from "@react-email/components";
+import { REPLY_EMAIL } from "./email/utils";
+import PendingMemberEmail from "@/emails/pending-member-email";
+SendGrid.setApiKey(process.env.SENDGRID_API_KEY);
+
+export interface SendConfirmationEmailProps {
+  email: string;
+  name: string;
+  location: string;
+  link: string;
+  recordID: string;
+  title?: string; // title is only non-required freeform field on sign up
+}
+
+export async function sendConfirmationEmails({
+  email,
+  recordID,
+  name,
+  location,
+  link,
+  title,
+}: SendConfirmationEmailProps) {
+  try {
+    // TODO better error handling, thrown in `create-member`
+    if (!recordID) throw new Error("No recordID provided");
+    if (!email) throw new Error("No email provided");
+    if (!name) throw new Error("No name provided");
+    if (!location) throw new Error("No location provided");
+    if (!link) throw new Error("No link provided");
```

Comment:
yep good for now but we could have something like this being added in the future: https://github.com/hawaiians/hawaiiansintech/commit/001c6552fd7c4a416793258c8346b4db1fb8276b#diff-742982ac7f59405221728f3fe3edf162c28f695dfebfada32146e5ba0141d299

Slowly been understanding that JS really doesn't throw errors by default for much

Reply:
Yeah, really liking the approach. Sort of left this lazily knowing that you put some thought into building out errors in your PR.

As long as we handle the error well in the upper-bound `try {} catch() {}` too (say if/when we want to error client-side). And we can discuss logging and all that in your PR.